src/main.cpp

```diff
@@ -2772,44 +2773,111 @@ void static UpdateTip(CBlockIndex *pindexNew) {
     if (!IsInitialBlockDownload())
     {
         int nUpgraded = 0;
+        bool upgradedEval = false;  // bip135 added
         const CBlockIndex* pindex = chainActive.Tip();
-        for (int bit = 0; bit < VERSIONBITS_NUM_BITS; bit++) {
-            WarningBitsConditionChecker checker(bit);
-            ThresholdState state = checker.GetStateFor(pindex, chainParams.GetConsensus(), warningcache[bit]);
-            if (state == THRESHOLD_ACTIVE || state == THRESHOLD_LOCKED_IN) {
-                if (state == THRESHOLD_ACTIVE) {
-                    strMiscWarning = strprintf(_("Warning: unknown new rules activated (versionbit %i)"), bit);
-                    if (!fWarned) {
-                        AlertNotify(strMiscWarning);
-                        fWarned = true;
+        // bip135 begin
```

Comment:
Why do you need to replace the warning code? Why not keep `WarningBitsConditionerChecker`. Also, this big chunk of code would be nice to have in its own function.